enclave/core/switchlesscalls.c

```diff
@@ -44,6 +53,12 @@ oe_result_t oe_handle_init_switchless(uint64_t arg_in)
     oe_switchless_call_manager_t safe_manager;
     size_t contexts_size, threads_size;
 
+    oe_spin_lock(&_lock);
+
+    // Check if already initialized
+    if (_host_worker_count != 0)
+        OE_RAISE(OE_ALREADY_EXISTS);
```

Comment:
OE_RAISE would bypass unlocking of the spinlock.
A coding pattern to handle this cleanly is:

```c
bool locked = oe_spin_lock(&_lock);
....
...

done:
    if(locked)
        oe_spin_unlock(&_lock);
```

Reply:
I think this error name `OE_ALREADY_EXISTS` is  ambiguous, probably don't fit into the scenario as well. Can we have a specific error code like `OE_INITIALIZED_MORE_THAN_ONCE`?

Reply:
I know that `OE_ALREADY_EXISTS` implies there was some resource being created but I think its clear given the context of the function being called (an init function) that OE_ALREADY_EXISTS means that switchless has already been initialized.

I understand your point, and we could add something like `OE_ALREADY_INITIALIZED`  but I don't want to create too many error codes that mean very similar things.

Reply:
Without reading the comments in result.h, I'd think `OE_ALREADY_EXIST` means the enclave already exists. I bet most people would think similarly.

Reply:
I think the error has to be taken into context of where it was thrown from (which verbose logging will tell you). Because even if we add `OE_ALREADY_INITIALIZED`, its not clear that it is referring to switchless calls unless you look at which function threw the error. You could similarly infer that the enclave itself has already been initialized.

Reply:
I am speaking from a readability perspective. Sorry for being picky. I just don't think `OE_ALREADY_EXIST` should be an error code name at the first place. `OE_ALREADY_INITIALIZED` is a much better name, and covers all scenarios that a singleton object is initialized twice.

Reply:
I agree that `OE_ALREADY_INITIALIZED` or even `OE_ALREADY` would be better names. The problem is that changing the name of an enum value is a breaking change in a public header and I don't know if this case warrants making that change.

At the very least if we are going to change the name of the enum I think it should be done in a different PR.

Reply:
I agree, it would be better not to make a breaking change.

Reply:
Please create an issue for the name changes.

Reply:
#2456